dicthash/dicthash.py

```diff
@@ -219,14 +228,24 @@ def validate_blackwhitelist(d, l):
             validate_blackwhitelist(d[key[0]], [key[1:]])
 
 
-def filter_blackwhitelist(l, key):
+@overload
+def filter_blackwhitelist(l: None, key: Optional[Hashable]) -> None:
+    pass
+
+
+@overload
+def filter_blackwhitelist(l: list, key: Optional[Hashable]) -> list:
+    pass
+
+
+def filter_blackwhitelist(l: Optional[list], key: Optional[Hashable]) -> Union[list, None]:
```

Comment:
could you please quickly explain what these constructs are?

Reply:
These constructs deal with the problem that the return value of `filter_blackwhitelist` is conditional on the type of `l` (see its docstring).
The type checker cannot perform proper type checking due to this conditional return values. Therefore, the solution is to use the `overload` decorator and add these auxiliary function definitions that just serve to clarify the conditional return values.
See https://stackoverflow.com/questions/44233913/is-there-a-way-to-specify-a-conditional-type-hint-in-python

Reply:
great, thanks!